python/node_classification.py

```diff
@@ -0,0 +1,421 @@
+        nodes, edges, global_params[DataParams.SPLIT_RATIO], global_params, reindexing
+    )
+
+    # print(data)
+    # print(data.edge_index.max())
+
+    global_params[ModelParams.IN_CHANNELS] = np.shape(
+        Modelling.data.x.detach().numpy()
+    )[1]
+
+    global_params[ModelParams.OUT_CHANNELS] = len(
+        set(Modelling.data.y.detach().numpy())
+    )
+
+    Modelling.model = InductiveModel(
+        layer_type=global_params[ModelParams.LAYER_TYPE],
+        in_channels=global_params[ModelParams.IN_CHANNELS],
+        hidden_features_size=global_params[ModelParams.HIDDEN_FEATURES_SIZE],
+        out_channels=global_params[ModelParams.OUT_CHANNELS],
+        aggr=global_params[ModelParams.AGGREGATOR],
+    )
+
+    # obtain function by its name from library
+    Modelling.opt = torch.optim.Adam(
+        Modelling.model.parameters(),
+        lr=global_params[OptimizerParams.LEARNING_RATE],
+        weight_decay=global_params[OptimizerParams.WEIGHT_DECAY],
+    )
+
+    Modelling.criterion = torch.nn.CrossEntropyLoss()
+
+
+@mgp.read_proc
+def set_model_parameters(
+    ctx: mgp.ProcCtx, params: mgp.Map = {}
+) -> mgp.Record(in_channels=int, out_channels=int, num_samples=int, path=str):
+    """The purpose of this function is to initialize all global variables. Parameter
+    params is used for variables written in query module. It first checks
+    if (new) variables in params are defined appropriately. If so, map of default
+    global parameters is overriden with user defined dictionary params.
+    After that it executes previously defined functions declare_globals and
+    declare_model_and_data and sets each global variable to some value.
+
+    Args:
+        ctx: (mgp.ProcCtx): current context,
+        params: (mgp.Map, optional): user defined parameters from query module. Defaults to {}
+
+    Raises:
+        Exception: exception is raised if some variable in dictionary params is not
+                    defined as it should be
+
+    Returns:
+        mgp.Record(
+            in_channels (int): number of input channels,
+            out_channels (int): number of output channels
+            num_sample(int): number of samples
+        )
+    """
+    global DEFINED_INPUT_TYPES, DEFAULT_VALUES
+
+    # function checks if input values in dictionary are correctly typed
+    def is_correctly_typed(defined_types, input_values):
+        if isinstance(defined_types, dict) and isinstance(input_values, dict):
+            # defined_types is a dict of types
+            return all(
+                k in input_values  # check if exists
+                and is_correctly_typed(
+                    defined_types[k], input_values[k]
+                )  # check for correct type
+                for k in defined_types
+            )
+        elif isinstance(defined_types, type):
+            return isinstance(input_values, defined_types)
+        else:
+            return False
+
+    # mgconsole bug
```

Comment:
Where is the bug?

Reply:
Sorry, I don't remember any more. There is no bug.

Reply:
Remove comment :smile: